client/components/domains/register-domain-step/index.jsx

```diff
@@ -31,19 +32,53 @@ var wpcom = require( 'lib/wp' ).undocumented(),
 	abtest = require( 'lib/abtest' ).abtest;
 
 // max amount of domain suggestions we should fetch/display
-var SUGGESTION_QUANTITY = 10,
+const SUGGESTION_QUANTITY = 10,
 	INITIAL_SUGGESTION_QUANTITY = 2;
 
+const analytics = analyticsMixin( 'registerDomain' );
+
+let searchQueue = [],
+	searchStackTimer = null;
+
+function processSearchQueue() {
+	var q = searchQueue.slice();
+	window.clearTimeout( searchStackTimer );
+	searchStackTimer = null;
+	searchQueue = [];
+
+	outerLoop:
+		for ( let i = 0; i < q.length; i++ ) {
+			for ( let k = i + 1; k < q.length; k++ ) {
+				if ( startsWith( q[ k ].query, q[ i ].query ) ) {
+					continue outerLoop;
+				}
+			}
+			reportSearch( q[ i ] );
+		}
+}
+
+function reportSearch( { query, section } ) {
+	analytics.recordEvent( 'searchFormSubmit', query, section );
+}
+
+function enqueueSearch( search ) {
+	searchQueue.push( search );
+	if ( searchStackTimer ) {
+		window.clearTimeout( searchStackTimer );
+	}
+	searchStackTimer = window.setTimeout( processSearchQueue, 10000 );
+}
+
 var RegisterDomainStep = React.createClass( {
-	mixins: [ analyticsMixin( 'registerDomain' ) ],
+	mixins: [ analytics ],
 
 	propTypes: {
 		cart: React.PropTypes.object,
 		onDomainsAvailabilityChange: React.PropTypes.func,
 		products: React.PropTypes.object.isRequired,
 		selectedSite: React.PropTypes.oneOfType( [ React.PropTypes.object, React.PropTypes.bool ] ),
 		basePath: React.PropTypes.string.isRequired,
-		suggestion: React.PropTypes.string,
+		suggestion: React.PropTypes.string
```

Comment:
That's why we have Babel - it removes the trailing comma :)
Also, from what I understand (thanks for the extra info about this inconsistency in JS's implementation across browsers!) it's a problem for IE8 which I don't think we support anyway :)
Anyway, sorry, I didn't mean to start a discussion here - I just didn't want you to introduce changes that are not related to the PR and are a matter of preference (I don't consider the trailing comma an error and there are no guidelines for/against them).